### ISSUE TITLE
Why my rtp video track published to livekit by server-sdk-go is not display on flutter

### ISSUE THREAD

**Issue by xiangxud** [livekit/server-sdk-go#81]
I use server-sdk-go create room and publish rtp track ,some h264 video can display ,but some h264 rtp packets not display on flutter sdk, my js on web is ok,how can i debug?
 
this is my published:

	if t.VideoRTPTrack == nil {
		if strings.Contains(trackRemote.Codec().MimeType, "video") {
			videoRTPTrack, err := webrtc.NewTrackLocalStaticRTP(trackRemote.Codec().RTPCodecCapability, streamname+"-video", streamname)
			if err != nil {
				panic(err)
			}

			// r.RoomClient.MutePublishedTrack(r.Ctx,)
			// var local_video *lksdk.LocalTrackPublication
			if t.Videopub, err = t.RoomConnect.LocalParticipant.PublishTrack(videoRTPTrack, &lksdk.TrackPublicationOptions{Name: streamname + "-video"}); err != nil {
				log.Debug("Error publishing video RTP track->", err)
				return err
			}
			t.VideoRTPTrack = videoRTPTrack
			r.Localtracks[streamname] = t
			// r.Localtracks[streamname] = &LocalTrackPublication{p: local_video, Track: videoTrack, Trackname: streamname + "-video"}
			log.Debug("published video track -> ", streamname)
		}
	}
	if t.AudioRTPTrack == nil {
		if strings.Contains(trackRemote.Codec().MimeType, "audio") {
			audioRTPTrack, err := webrtc.NewTrackLocalStaticRTP(trackRemote.Codec().RTPCodecCapability, streamname+"-audio", streamname)
			//audioTrack, err := lksdk.NewLocalSampleTrack(webrtc.RTPCodecCapability{MimeType: webrtc.MimeTypeOpus})
			if err != nil {
				panic(err)
			}
			// var local_audio *lksdk.LocalTrackPublication

			if t.Audiopub, err = t.RoomConnect.LocalParticipant.PublishTrack(audioRTPTrack, &lksdk.TrackPublicationOptions{Name: streamname + "-audio"}); err != nil {
				log.Debug("Error publishing audio track", err)
				return err
			}
			t.AudioRTPTrack = audioRTPTrack
			r.Localtracks[streamname] = t
			log.Debug("published audio track -> ", streamname)
		}
	}

this is write rtp data:

func (r *Room) TrackSendRtpPackets(trackname, kind string, data []byte) (n int, err error) {
	if trackname == "" {
		log.Debug("Track name is null")
		return 0, fmt.Errorf("input trackname is null")
	}
	// var t *webrtc.TrackLocalStaticSample
	var t *webrtc.TrackLocalStaticRTP
	track := r.Localtracks[trackname]
	if track == nil {
		log.Debug("Track is nil ->", trackname, "<- no to publish")
		return 0, fmt.Errorf(" track is null,no to publish")
	}
	if kind == "video" {
		t = track.VideoRTPTrack
	} else if kind == "audio" {
		t = track.AudioRTPTrack
	}
	if t == nil {
		log.Debug("Track is nil ->", trackname, "<- no to publish")
		return 0, fmt.Errorf(" track is null,no to publish")
	}

	n, err = t.Write(data)
	return n, err

}

**Comment by davidzhao** [comment]
which platform are you testing with for Flutter? It's probably due to the profile used for H.264. WebRTC spec only supports constrained baseline; While some browsers may render other profiles fine, mobile devices can be very limiting when it comes to decoding support.

Ensure your H.264 video is encoded [like this](https://github.com/livekit/server-sdk-go#h264--opus)

**Comment by xiangxud** [comment]
thx,flutter for web edge ,i use h264 profile is 77 (main) not 66 baseline, let me try baseline

**Comment by xiangxud** [comment]
always only audio play ,not video play,i set config my device h264 encode profile to baseline

**Comment by davidzhao** [comment]
what do the client logs say? can you check chrome://webrtc-internals to see if the track exists?

**Comment by xiangxud** [comment]
yes the track exists ,but not frame received ,the audio track is ok

**Comment by davidzhao** [comment]
Unfortunately, we aren't able to reproduce this. Are you able to give us a small example that could reproduce this? Ideally with our flutter example app.
